### PR TITLE
feat: skeleton loading feature added as support

### PR DESCRIPTION
**Pull request by yashsehgal**
- Skeleton loading feature added ✅ 
- Component support created as a utility with type support ✅ 
- Skeleton layout and styling ⚠️

### CODE REVIEW

**Comment by yashsehgal** [comment]
- Without Skeleton Loader

Screenshot 2022-12-13 at 6 03 14 AM

- With Skeleton Loader

Screenshot 2022-12-13 at 6 02 20 AM

File: components/ui-patterns/Skeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { CardSkeletonType } from '../../types/ui-pattern-types/skeleton-types';
+
+function CardSkeleton({ visibility }: CardSkeletonType) {
+    return (
+        <Skeleton 
+            width="280px"
+            height="320px"
+            style={{
+                opacity: visibility
+            }}
```
Comment (ayushsoni1010) [inline components/ui-patterns/Skeleton.tsx]: Hey @yashsehgal 👋

Do we have other props for the `Skeleton Wrapper Component` which are provided by the imported library
if yes, then

```
<Skeleton 
    width="280px"
    height="320px"
    style={{
      opacity: visibility
    }}
    {...props}
/>
```

Comment (yashsehgal) [reply components/ui-patterns/Skeleton.tsx]: Yes, this is a nice add-on to this component, if you want you can add this in the `type`. ✅

Comment (ayushsoni1010) [reply components/ui-patterns/Skeleton.tsx]: As you are currently working on it, can you make the fix for this?

Required changes:
- UX of the skeleton loading is not up to the point, just showing a square doesn't make a good UX flow
- Let's build the skeleton loading based on its data, like if we have images, headings, and descriptions, etc. So we have to make the UI similar to these components.

Comment (yashsehgal) [reply components/ui-patterns/Skeleton.tsx]: Sure @ayushsoni1010 👍🏽 I am currently working on this change only.

> Let's build the skeleton loading based on its data, like if we have images, headings, and descriptions, etc. So we have to make the UI similar to these components.

For this, I will making component wrappers, according to different use-cases.
Just like this `<EventCardSkeleton />`, `<TeamCardSkeleton />` and etc

**Comment by yashsehgal** [comment]
Merge Conflicts from branch `feat/skeleton-loading` are resolved, generated from master after merging changes from [`refactor/config-imports-paths`](https://github.com/hekors/website/commit/091c7bc89c286eeec43bdbd80347021dbf4ac19e)

File: components/ui-patterns/Skeleton.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import Skeleton, { SkeletonTheme } from 'react-loading-skeleton'
+import 'react-loading-skeleton/dist/skeleton.css'
+import { CardSkeletonType } from '../../types/ui-pattern-types/skeleton-types';
```
Comment (ayushsoni1010) [inline components/ui-patterns/Skeleton.tsx]: can you fix the import pattern?

Also, do share the skeleton UI, it would be helpful

**Comment by yashsehgal** [comment]
I have added skeleton card wrappers for all the sections in the website.

### Upcoming Hackathons Section
Screenshot 2022-12-16 at 2 07 10 AM

### Upcoming Sessions Section
Screenshot 2022-12-16 at 2 07 21 AM

### Team Members Section
Screenshot 2022-12-16 at 2 07 46 AM

### Community Narratives Section
Screenshot 2022-12-16 at 2 08 01 AM

**Comment by ayushsoni1010** [comment]
Hey @yashsehgal 

Good work! Skeleton loading UI's are really pretty, we just need to enhance the UX of the `Upcoming Sessions Section` card
It seems a little bit odd, isn't it?

Let's put the skeleton UI for `image` as well, below skeleton is good, as you have done for the `team members` skeleton UI
P.S: Do not make the below skeleton UI the same for both cards, we are mainly focusing only on the image part.
I hope you got it right

Let me know if we can enhance it

**Review by ayushsoni1010 — CHANGES_REQUESTED** [review]
Let's go with the import pattern
Prioritized with numbering in order to follow the pattern

1. React, Next: Basic Imports
2. Any library used: Library Imports
3. Any file or components: Components Imports
4. For styles: Styles Imports
5. For types: Types imports
6. For icons: Icons Imports

Let me know if any changes we have to do in this pattern or any thoughts on this.

**Comment by yashsehgal** [comment]
> Let's go with the import pattern Prioritized with numbering in order to follow the pattern
> 
> 1. React, Next: Basic Imports
> 2. Any library used: Library Imports
> 3. Any file or components: Components Imports
> 4. For styles: Styles Imports
> 5. For types: Types imports
> 6. For icons: Icons Imports
> 
> Let me know if any changes we have to do in this pattern or any thoughts on this.

This sounds like a good modification to me, will make sure these imports.

**Comment by yashsehgal** [comment]
> Hey @yashsehgal
> 
> Good work! Skeleton loading UI's are really pretty, we just need to enhance the UX of the `Upcoming Sessions Section` card It seems a little bit odd, isn't it?
> 
> Let's put the skeleton UI for `image` as well, below skeleton is good, as you have done for the `team members` skeleton UI P.S: Do not make the below skeleton UI the same for both cards, we are mainly focusing only on the image part. I hope you got it right
> 
> Let me know if we can enhance it

Got what you're trying to say here, working on it ✅

**Comment by yashsehgal** [comment]
> Let's go with the import pattern Prioritized with numbering in order to follow the pattern
> 
> 1. React, Next: Basic Imports
> 2. Any library used: Library Imports
> 3. Any file or components: Components Imports
> 4. For styles: Styles Imports
> 5. For types: Types imports
> 6. For icons: Icons Imports
> 
> Let me know if any changes we have to do in this pattern or any thoughts on this.

Can we put this task into an other sub-task, as I need to publish this PR to the master for deployment/testing purposes.

**Pull request merged**